keras/engine/topology.py

```diff
@@ -413,7 +415,10 @@ def add_weight(self,
                             name=name,
                             constraint=constraint)
         if regularizer is not None:
-            self.add_loss(regularizer(weight))
+            reg_tensor = regularizer(weight)
+            reg_tensor._weight_id = Layer._weight_id
+            Layer._weight_id += 1
```

Comment:
This is not thread-safe.

Reply:
Ok, I completely misunderstantood what you meant earlier with the use of id(), my bad. I'll change it and use set().